iqama_fetcher.py

```diff
@@ -1,42 +1,122 @@
+  if not masjid_id:
+    return None
+
+  request = requests.get(_IQAMAH_NET_URL + str(masjid_id) + ".xml", timeout=15)
 
   # checks if request isn't bad
   # pylint: disable=no-member
   if request.status_code == requests.codes.ok:
-    response = json.loads(request.text)
-    print 'response from scraper service = ', response
-
-    # translate the dict keys to enums
-    iqama_times = {}
-    for key in response:
-      prayer = util.StringToDailyPrayer(key)
-      if prayer:
-        iqama_times[prayer] = response[key]
-
-    return iqama_times[desired_prayer]
-  print 'bad response from scraper service = ', request.status_code
+    # setting a tree node to the beginning of request xml content
+    tree = ElementTree.fromstring(request.content)
+    # setting an element node to element 'IqDay' which contains the
+    # prayer names and timings
+    # <IqDay><Fajr>##:##</Fajr>.....</IqDay>
+    iqamah_element = tree.findall('IqDay')[0]
+
+    # looping through the IqDay tags (prayer names)
+    for iqamah in iqamah_element:
+      # convert the prayer names in the IqDay element to a known format
+      iqamah_name = GetEquivalentPrayer(str(iqamah.tag))
+      if desired_prayer.lower() == (iqamah_name).lower():
+        iqama_time = iqamah.text
+        break
+
+    print 'Iqama Time ', iqama_time
+
+    # if iqama time was found then format time to ##:##
+    if iqama_time:
+      return FormatTime(str(iqama_time))
+    return None
+  print 'bad response from iqamah service = ', request.status_code
   return None
+
+def FormatTime(time):
+  """Returns the time variable back in ##:## format
+
+  Args:
+    time: a string representing the iqamah time
+
+  Returns: a string containing the iqamah time in ##:## format
+  """
+
+  # if the time input is in a format where it contains integers
+  # then proceed forward otherwise return None
+  if re.match(r'(\d+)', time, flags=0):
+    # strip the time variable from anything added to it
+    # example: 10:00 PM becomes 1000 and 5 AM becomes 5
+    time = ''.join(re.findall(r'(\d+)', time))
+    # if time length is more than 2 characters then add a : after
+    # the second character from the right
+    # example: 1000 becomes 10:00 and 5 stays 5
+    if len(time) > 2:
+      return time[:1] + ':' + time[1:] 
+    return time + ':00' 
+
+def GetEquivalentPrayer(prayer):
+  """Returns the prayer equivalent to the defined prayer-name entity
+
+  Args:
+    prayer: a string representing desired salah
+
+  Returns: a string containing prayer-name entity equivalent to the desired
+  input prayer
+  """
+  prayer = prayer.lower()
+  eq_prayer = None
+  
+  _FAJR = re.compile(r'((^f)(.+)(r$))')
+  _DHUHR = re.compile(r'((^d)(.+)(r$))|((^z)(.+)(r$))')
+  _ASR = re.compile(r'((^a)(.+)(r$))')
+  _MAGHRIB = re.compile(r'((^m)(.+)(b$))')
+  _ISHA = re.compile(r'((^i)(.+)(a$))')
+  _JUMMA = re.compile(r'((^g))|((^j))')
+  
+  # if prayer matches an 'f' at the beginning and an 'r' at the end
```

Comment:
it probably makes more sense to put these comments alongside the definition of the regex